vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -910,6 +914,7 @@ import { isFocusable } from '@vaadin/grid/src/vaadin-grid-active-item-mixin.js';
         });
 
         grid.$connector.confirm = tryCatchWrapper(function (id) {
+          grid.$connector.confirmingParentItems = true;
```

Comment:
I'll probably extract this to a separate PR since it's somewhat unrelated. An ideal fix would be to make sure `expandedItems` are assigned _before_ `confirm` executes.